### PR TITLE
Overhaul system objects infrastructure

### PR DESCRIPTION
**Pull request by tfcollins**
This PR simplifies much of the core code and enables better support for:
- DDSs
- Non-complex data devices
- Multichannel devices

Tests have been added for multichannel configurations

### CODE REVIEW

**Comment by tfcollins** [comment]
Need to add tests for AD9371 OBS receiver before merge

File: hil_models/ad9371/ad9371_mimo.m [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    while ~valid
+        [out, valid] = rx();
+    end
+end
+rx.release();
+tx.release();
+
+%% Plot
+nSamp = length(out);
+fs = tx.SamplingRate;
```
Comment (SrikanthPagadarai) [inline hil_models/ad9371/ad9371_mimo.m]: Shouldn't this be `fs = swv1.SampleRate;`

File: hil_models/ad9371/ad9371_mimo.m [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+FFTRxData1  = fftshift(10*log10(abs(fft(out(:,1)))));
+FFTRxData2  = fftshift(10*log10(abs(fft(out(:,2)))));
+df = fs/nSamp;  freqRangeRx = (-fs/2:df:fs/2-df).'/1000;
```
Comment (SrikanthPagadarai) [inline hil_models/ad9371/ad9371_mimo.m]: Here, if we have `freqRangeRx = (-fs/2:df:fs/2-df).'/1e6;`, the, two lines later, we can have ...

File: hil_models/ad9371/ad9371_mimo.m [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+subplot(2,1,1);
+plot(freqRangeRx, FFTRxData1);
+xlabel('Frequency (kHz)');ylabel('Amplitude (dB)');grid on;
```
Comment (SrikanthPagadarai) [inline hil_models/ad9371/ad9371_mimo.m]: `xlabel('Frequency (MHz)');`. Similarly, for the second plot...

File: hil_models/ad9371/ad9371_mimo.m [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+subplot(2,1,2);
+plot(freqRangeRx, FFTRxData2);
+xlabel('Frequency (kHz)');ylabel('Amplitude (dB)');grid on;
```
Comment (SrikanthPagadarai) [inline hil_models/ad9371/ad9371_mimo.m]: `xlabel('Frequency (MHz)');`

File: hil_models/ad9371/ad9371_mimo.m (hunk 1 above)
Comment (tfcollins) [reply hil_models/ad9371/ad9371_mimo.m]: Is this value different than tx.SamplingRate?

Comment (SrikanthPagadarai) [reply hil_models/ad9371/ad9371_mimo.m]: Yes.

**Pull request merged**